src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -1954,6 +1980,30 @@ void Session::send_json_error_message()
 }
 
 
+void Session::send_test_command_message()
+{
+    REALM_ASSERT(m_state == Active);
+
+    auto it = std::find_if(m_pending_test_commands.begin(), m_pending_test_commands.end(),
+                           [](const PendingTestCommand& command) {
+                               return !command.body.empty();
+                           });
+    REALM_ASSERT(it != m_pending_test_commands.end());
+
+    ClientProtocol& protocol = m_conn.get_client_protocol();
+    OutputBuffer& out = m_conn.get_output_buffer();
+    auto session_ident = get_ident();
+
+    logger.info("Sending: TEST_COMMAND \"%1\" (session_ident=%2, request_ident=%3)", it->body, session_ident, it->id);
+    protocol.make_test_command_message(out, session_ident, it->id, it->body);
+
+    m_conn.initiate_write_message(out, this); // Throws;
+    it->body.clear();
```

Comment:
It seems weird to clear the message body to indicate it's been sent, I would probably prefer a "processed" or "sent" flag/state value. Although it _is_ better use of memory to clear it once you don't need it anymore...

Reply:
`clear()` doesn't actually have to release any memory, it just changes the size to zero. see https://en.cppreference.com/w/cpp/string/basic_string/clear. I've added a `pending` flag though that we use instead of seeing if the body is empty.